### PR TITLE
Convert ah commands to rzshell

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- `rz_core_analysis_hint_list()` was renamed to `rz_core_analysis_hint_list_print()` 
- `rz_core_analysis_hint_list_print()` and `rz_core_analysis_hint_print()` use `RzCmdStateOutput`
- all commands work only with `core->offset`
- `ah` was replaced with `ahl`
- `ah.` and `ah <addr>` were replaced with `ahl.`
- missing `aho-` and `aht-` were added

**Test plan**

...

**Closing issues**

Related: #1442
...

### CODE REVIEW

**Comment by XVilka** [comment]
Maybe it worth to address also the https://github.com/rizinorg/rizin/issues/752 while you are changing this code.

**Comment by p371jah** [comment]
I guess #752 requires unifying of all disasm plugins.

**Comment by ret2libc** [comment]
> Maybe it worth to address also the #752 while you are changing this code.

Completely separate problem I think. out of scope for this rzshell conversion IMO.

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -1258,7 +1258,7 @@ static void print_hint_h_format(HintNode *node) {
 // if mode == 'j', pj must be an existing PJ!
 static void hint_node_print(HintNode *node, int mode, PJ *pj) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: Maybe switch the function argument to `RzOutputMode` enum?

**Review by XVilka — APPROVED** [review]
LGTM apart that minor thing.

**Comment by p371jah** [comment]
I have ported the code but I don't know what to use as a summary for commands. Previously there was no description in the help,  but only command examples. Well I think I should move these examples to `yaml`.

**Comment by ret2libc** [comment]
> I have ported the code but I don't know what to use as a summary for commands. Previously there was no description in the help, but only command examples. Well I think I should move these examples to `yaml`.

Yes move the examples there please. For the actual text, let's not worry too much at this time. We will improve the descriptions as we see them.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@ret2libc take one more look, please, apart from that it's good to merge, I think.

**Review by wargio — APPROVED** [review]
LGTM !

File: librz/core/cmd/cmd_analysis.c [hunk 2]
```diff
@@ -8905,3 +8542,268 @@ RZ_IPI RzCmdStatus rz_rzil_vm_status_handler(RzCore *core, int argc, const char
...
+RZ_IPI RzCmdStatus rz_analysis_hint_del_offset_handler(RzCore *core, int argc, const char **argv) {
+	rz_analysis_hint_unset_offset(core->analysis, core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_list_struct_offsets_handler(RzCore *core, int argc, const char **argv) {
+	ut64 toff = rz_num_math(core->num, argv[1]);
+	if (!toff) {
+		return RZ_CMD_STATUS_WRONG_ARGS;
+	}
```
Comment (ret2libc) [inline librz/core/cmd/cmd_analysis.c:8799]: I think this should be replaced with `rz_num_is_valid_input`

**Review by ret2libc — COMMENTED** [review]
LGTM apart from that small thing with rznum.

File: librz/core/cmd/cmd_analysis.c (hunk 2 above)
Comment (p371jah) [reply librz/core/cmd/cmd_analysis.c:8799]: 1) About check on zero: zero offset is not allowed.
2) About `rz_num_is_valid_input`: I expect that rzshell validates arguments of type `RZ_CMD_ARG_TYPE_NUM`/`RZ_CMD_ARG_TYPE_RZNUM`

Comment (ret2libc) [reply librz/core/cmd/cmd_analysis.c:8799]: > About check on zero: zero offset is not allowed

Ah ok

> About rz_num_is_valid_input: I expect that rzshell validates arguments of type RZ_CMD_ARG_TYPE_NUM/RZ_CMD_ARG_TYPE_RZNUM

This is a good point actually. It is not done at the moment, but maybe it should be done indeed.

**Review by ret2libc — DISMISSED** [review]

File: librz/core/cmd/cmd_analysis.c (hunk 2 above)
Comment (ret2libc) [reply librz/core/cmd/cmd_analysis.c:8799]: Mmmm why zero offset is not allowed by the way?

Comment (p371jah) [reply librz/core/cmd/cmd_analysis.c:8799]: Because all structs have field at zero offset.

**Pull request merged**